qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -782,15 +782,15 @@ def _replace_parameterized_gate(op):
         def is_supercontrolled(gate):
             try:
                 operator = Operator(gate)
-            except QiskitError:
+            except (QiskitError, TypeError):  # TypeError: parametrized gate
```

Comment:
What is the source of a `TypeError` here? Is it coming from `__array__` ? 

I ask because I'm wondering if this is the correct fix, if it's coming from `__array__`  because the gate is parameterized in the target, I'm wondering if we should instantiate it with a fixed angle instead. Basically update `_replace_parameterized_gate` to handle `RZZ` too, or instead maybe try to generalize it. A parameterized gate in the target represents that any angle value is valid input so we're free to evaluate the matrix at a fixed angle if we want.